beater/otlp/grpc.go

```diff
@@ -129,7 +91,15 @@ func collectMetricsMonitoring(mode monitoring.Mode, V monitoring.Visitor) {
 	currentMonitoredConsumerMu.RLock()
 	c := currentMonitoredConsumer
 	currentMonitoredConsumerMu.RUnlock()
-	if c != nil {
-		c.collectMetricsMonitoring(mode, V)
+	if c == nil {
+		return
 	}
+
+	V.OnRegistryStart()
+	V.OnRegistryFinished()
```

Comment:
```suggestion
	defer V.OnRegistryFinished()
```

... the old code was wrong, not sure how it was passing tests. Maybe it doesn't matter :shrug: